### DISCUSSION TITLE
Compile and configuration problems with native mod_proxy_cluster modules

### DISCUSSION THREAD

**Discussion by ErhardSiegl** [modcluster/mod_cluster#655] [Q&A] [answered]
Hi,
After quite some time not using the apache modules for mod_cluster, I decided to try it on Ubuntu. Since I didn't find binaries, I checked out mod_proxy_cluster and tried to compile it ( commit 916629574dc208efb074ef9b79555c002207839c) Got this error: 
```
/home/ubuntu/src/mod_proxy_cluster/native/mod_proxy_cluster/mod_proxy_cluster.c: In function ‘proxy_cluster_trans’:
/home/ubuntu/src/mod_proxy_cluster/native/mod_proxy_cluster/mod_proxy_cluster.c:2223:46: error: ‘PROXYPASS_MAP_ENCODED’ undeclared (first use in this function)
 2223 |             int enc = (dconf->alias->flags & PROXYPASS_MAP_ENCODED) != 0;
      |                                              ^~~~~~~~~~~~~~~~~~~~~
```
It seems PROXYPASS_MAP_ENCODED was introduced in https://github.com/modcluster/mod_proxy_cluster/commit/ca06df1085f2477f274542410896209846fb31d2

As workaround I defined
`#define PROXYPASS_MAP_ENCODED 0` in `mod_proxy_cluster.c` which fixed the error.

I installed the modules and tried the config from https://docs.modcluster.io/#quick-start-guide.
With this the manager_module didn't load because of
```
 [:error] [pid 195880:tid 140018130123840] manager_init: apr_file_open for lock failed
```
The line
```
MemManagerFile /var/cache/apache2/mod_cluster
LoadModule proxy_module /usr/lib/apache2/modules/mod_proxy.so
...
```
and creating a directory `/var/cache/apache2/mod_cluster` fixed the problem. 
The rest was as expected.

Maybe you could add a hint to the (correct) solution to the quick-start-guide or a little more verbose error message in the code.

Kind regards
Erhard

**Comment by jfclere** [comment, marked as answer]
PROXYPASS_MAP_ENCODED is defined in mod_proxy.h
[jfclere@fedora proxy]$ grep PROXYPASS_MAP_ENCODED *.h
mod_proxy.h:#define PROXYPASS_MAP_ENCODED 0x08

For the 
[:error] [pid 195880:tid 140018130123840] manager_init: apr_file_open for lock failed
Yes I will add the filename next time I am making a PR, feel free to create an issue if you want ;-)

**Reply by ErhardSiegl** [reply]
Thank you! Apparently my mod_proxy.h was too old.
